### PR TITLE
Custom zoom registration form

### LINKED ISSUE #15: zoom registration worker
It should be pretty similar to the mailing list signup one, except we'll need to use JWT for auth.

### PR DESCRIPTION
**Pull request by everthemore**
Update cloudflare worker to also be able to handle zoom registrations. 
TODO:

- [x] Update mailing list action endpoint (include 'mailinglist' in URL to differentiate post requests)
- [x] Auto generate the form on the client side, and include (hidden?) input field for zoom MeetingID.

Closes #15

### CODE REVIEW

**Comment by akhmerov** [comment]
The plan sounds good. If the meeting is running, the client side should also immediately open the join URL (adding this for later).

**Comment by everthemore** [comment]
If you get a chance @akhmerov, take a look at this workflow ('wranger dev' from top directory will start a local version to test).  

- localhost/LRC is meant to represent the website with upcoming talks. Clicking the button...
- ...asks the cloudflare worker to render a registration form via POST, passing along talk info (like speaker name and meeting ID)
- That form then POSTs the form data to the respective Zoom API endpoint

**Comment by everthemore** [comment]
TODO: Add boolean flag for Zoom yo notify registrant rather than custom email

**Review by akhmerov — CHANGES_REQUESTED** [review]
- [x] We should remove the form body from here (it lives in the other repo)
- [x] For simplicity let's use the default zoom registration emails
- [x] Let's sign people up for the mailing list right after they register; this requires also sending the event type with the form.
- [x] **Important** immediately after merging we should test whether mailing list signups still work.

**Comment by akhmerov** [comment]
In order to minimize disruption of the mailing list signups, let's check whether we can deploy 2 workers.

**Comment by everthemore** [comment]
> [ ] For simplicity let's use the default zoom registration emails

The [registration via the API](https://marketplace.zoom.us/docs/api-reference/zoom-api/meetings/meetingregistrantcreate) has the auto_approve flag that we can test.

**Comment by akhmerov** [comment]
> The registration via the API has the auto_approve flag that we can test.

I remembered wrong. Whether an email is sent or not, is actually meeting configuration, and it's already correctly set to `true`. So this requires no action.

**Comment by everthemore** [comment]
> [x] **Important** immediately after merging we should test whether mailing list signups still work.

Can confirm that this still works. The cloudflare worker now has a separate endpoint for mailing list signups, and one for zoom registrations that we can test separately by creating mock talks.

File: cloudflare/index.js [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+    var mailgunListName = "";
+    switch( eventtype ) {
+      case "lrc":
+        mailgunListName = "vsf-announce"
+        break;
+      case "speakers_corner":
+        mailgunListName = "speakers_corner"
+        break;
+      default:
+        return new Response("The list " + eventtype + " cannot be subscribed to.", {status:400, headers:corsHeaders})
```
Comment (akhmerov) [inline cloudflare/index.js]: Returning over here would result in an error message even though the registration goes through. We should either append to the result or let this slide.

**Review by akhmerov — APPROVED** [review]
I reviewed the code. Everything looks good except for the last minor problem below. We can also merge and test right away, and fix it separately later. That is perhaps a better approach.

**Comment by akhmerov** [comment]
I propose:

1. Merge this
2. Immediately push an update pointing the mailing list subscription to the new URL (separately from the other form updates)
3. Test whether mailing list subscription still works
4. Proceed with other improvements without time pressure.

File: cloudflare/index.js (hunk 1 above)
Comment (everthemore) [reply cloudflare/index.js]: Do you have a preference?

Comment (akhmerov) [reply cloudflare/index.js]: Adding to the response message "Unable to sign up to the mailing list, please report the problem to the VSF." is the best.

**Pull request merged**